uniffi_bindgen/src/interface/function.rs

```diff
@@ -89,13 +89,76 @@ impl Function {
     }
 
     pub fn derive_ffi_func(&mut self, ci_prefix: &str) -> Result<()> {
-        self.ffi_func.name = format!("{}_{}", ci_prefix, self.name);
+        // The name is already set if the function is defined in a proc-macro-generated JSON file
+        // rather than in UDL. Don't overwrite it in that case.
+        if self.ffi_func.name.is_empty() {
+            self.ffi_func.name = format!("{}_{}", ci_prefix, self.name);
+        }
+
         self.ffi_func.arguments = self.arguments.iter().map(|arg| arg.into()).collect();
         self.ffi_func.return_type = self.return_type.as_ref().map(|rt| rt.into());
         Ok(())
     }
 }
 
+impl From<uniffi_meta::FnParamMetadata> for Argument {
+    fn from(meta: uniffi_meta::FnParamMetadata) -> Self {
+        Argument {
+            name: meta.name,
+            type_: convert_type(&meta.ty),
+            by_ref: false,
+            optional: false,
+            default: None,
+        }
+    }
+}
```

Comment:
My thinking was that this would be a good starting point that would work well while the UDL stuff still exists as well. I think it makes sense to remove the other types after the proc-macro has fully replaced UDL.